src/test/java/core/basesyntax/service/impl/ReadDataServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+
+class ReadDataServiceImplTest {
+    private static final String SOURCE_DATA_FILE_PATH =
+            "src/test/resources/inputDataTest.csv";
+    private static final String FILE_NAME = "inputDataTest.csv";
+    private static final String EMPTY_FILE_NAME = "inputDataTestEmpty.csv";
+    private static final String NON_EXISTED_FILE_PATH = "";
+    private static final String EMPTY_DATA_TEST_FILE_PATH =
+            "src/test/resources/inputDataTestEmpty.csv";
+
+    private DataValidator dataValidatorMock;
+    private ReadDataService readDataService;
+    private List<String> testData;
+    private File file;
+    private BufferedReader bufferedReaderMock;
+    private List<String> expectedData;
+
+    @BeforeEach
+    void setUp() {
+        dataValidatorMock = mock(DataValidator.class);
+        file = mock(File.class);
+        bufferedReaderMock = mock(BufferedReader.class);
+        readDataService = new ReadDataServiceImpl(dataValidatorMock);
+        testData = Arrays.asList(
+                "type,fruit,quantity",
+                "b,banana,20",
+                "b,apple,100",
+                "s,banana,100",
+                "p,banana,13",
+                "r,apple,10",
+                "p,apple,20",
+                "p,banana,5",
+                "s,banana,50"
+        );
+    }
+
+    @Test
+    void readData_ValidFile_ok() throws Exception {
+        when(bufferedReaderMock.readLine()).thenReturn("b", "apple", "100", null);
+        when(file.getName()).thenReturn(FILE_NAME);
+        whenNew(File.class).withArguments(SOURCE_DATA_FILE_PATH).thenReturn(file);
+        doNothing().when(dataValidatorMock).validateIsSourceDataEmpty(anyList());
+        List<String> result = readDataService.readData(SOURCE_DATA_FILE_PATH);
+        List<String> trimmedTestData = trimListElements(testData);
+        List<String> trimmedResult = trimListElements(result);
+        assertEquals(trimmedTestData, trimmedResult);
+        verify(dataValidatorMock).validateIsSourceDataEmpty(anyList());
+    }
+
+    @Test
+    void readData_EmptyFile_notOk() throws Exception {
+        expectedData = Collections.emptyList();
+        when(bufferedReaderMock.readLine()).thenReturn(null);
+        when(file.getName()).thenReturn(EMPTY_FILE_NAME);
+        whenNew(File.class).withArguments(EMPTY_DATA_TEST_FILE_PATH).thenReturn(file);
+        doNothing().when(dataValidatorMock).validateIsSourceDataEmpty(anyList());
+        List<String> result = readDataService.readData(EMPTY_DATA_TEST_FILE_PATH);
+        assertEquals(expectedData, result);
+        verify(dataValidatorMock).validateIsSourceDataEmpty(anyList());
+    }
+
+    @Test
+    void readData_InvalidFile_notOk() throws Exception {
+        when(file.getName()).thenReturn(FILE_NAME);
+        whenNew(File.class).withArguments(NON_EXISTED_FILE_PATH).thenReturn(file);
+        whenNew(BufferedReader.class).withAnyArguments()
+                .thenThrow(new IOException("File not found"));
```

Comment:
powermock is a bad practice. You create a buffered stream in you code, and it's a part of inner logic, so don't mock it